src/main/java/apoc/number/Number.java

```diff
@@ -0,0 +1,126 @@
+import java.text.DecimalFormatSymbols;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.stream.Stream;
+
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+
+import apoc.Description;
+import apoc.result.DoubleResult;
+import apoc.result.LongResult;
+import apoc.result.StringResult;
+
+/**
+ * 
+ * @since 25.8.2016
+ * @author inserpio
+ */
+public class Number {
+
+  @Procedure
+  @Description("apoc.number.formatLong(number, pattern, locale) yield value | format a long using a (optional) pattern and for a (optional) locale to produce a string")
+  public Stream<StringResult> formatLong(final @Name("number") long number, @Name("pattern") String pattern, @Name("language") String language) {
+    DecimalFormat format = buildFormatter(pattern, language);
+    return Stream.of(new StringResult(format.format(number)));
+  }
+
+  @Procedure
+  @Description("apoc.number.formatDouble(number, pattern, locale) yield value | format a double using a (optional) pattern and for a (optional) locale to produce a string")
+  public Stream<StringResult> formatDouble(final @Name("number") double number, @Name("pattern") String pattern, @Name("language") String language) {
+    DecimalFormat format = buildFormatter(pattern, language);
+    return Stream.of(new StringResult(format.format(number)));
+  }
+  
+  @Procedure
+  @Description("apoc.number.parseAsLong(text, pattern, locale) yield value | parse a text using a (optional) pattern and for a (optional) locale to produce a long")
+  public Stream<LongResult> parseAsLong(final @Name("text") String text, @Name("pattern") String pattern, @Name("language") String language) throws ParseException {
+    DecimalFormat format = buildFormatter(pattern, language);
+    return Stream.of(new LongResult(format.parse(text).longValue()));
+  }
+
+  @Procedure
+  @Description("apoc.number.parseAsDouble(text, pattern, locale) yield value | parse a text using a (optional) pattern and for a (optional) locale to produce a double")
+  public Stream<DoubleResult> parseAsDouble(final @Name("text") String text, @Name("pattern") String pattern, @Name("language") String language) throws ParseException {
+    DecimalFormat format = buildFormatter(pattern, language);
+     return Stream.of(new DoubleResult(format.parse(text).doubleValue()));
+  }
+
+  @Procedure
+  @Description("apoc.number.negativePrefix(pattern, locale) yield value | get the negative prefix for the given (optional) pattern and (optional) locale")
+  public Stream<StringResult> negativePrefix(@Name("pattern") String pattern, @Name("language") String language) {
+    DecimalFormat format = buildFormatter(pattern, language);
+    return Stream.of(new StringResult(format.getNegativePrefix()));
+  }
+  
+  @Procedure
+  @Description("apoc.number.positivePrefix(pattern, locale) yield value | get the positive prefix for the given (optional) pattern and (optional) locale")
```

Comment:
not sure we need all these helper methods. I think if someone needs this information they can just try it out? With a concrete negative number or number > 1000 ?